services/localizely/localizely-lang-progress.service.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const { InvalidResponse } = require('..')
+const { coveragePercentage } = require('../color-formatters')
+const {
+  BaseLocalizelyService,
+  keywords,
+  documentation,
+  queryParamSchema,
+} = require('./localizely-base')
+
+module.exports = class LocalizelyLangProgress extends BaseLocalizelyService {
```

Comment:
Does this really need to be a separate service class? Seems like we could have a single service class that has `languageCode` as an optional param, and then have logic (probably in a new `transform` function) that will extract correct `reviewedProgress` value in kind, and a similar logic check in the `render` function to set the label accordingly

Something like this perhaps:
```js
  static render({ langName, reviewedProgress }) {
    return {
      label: langName || 'localized',
      message: `${reviewedProgress}%`,
      color: coveragePercentage(reviewedProgress),
    }
  }

  static transform(json, languageCode) {
    if (!languageCode) {
      return { reviewedProgress: json.reviewedProgress } 
    }
    
    const language = json.languages.find(lang => lang.langCode === languageCode)
    if (!language) {
      throw new InvalidResponse({ prettyMessage: 'Unsupported language' })
    }

    const { langName, reviewedProgress } = language
    return { langName,  reviewedProgress }
  }

  async handle({ projectId, branch }, { token: apiToken, languageCode }) {
    const json = await this.fetch({ projectId, branch, apiToken })
    const { langName, reviewedProgress } = this.constructor.transform(json, languageCode)
    return this.constructor.render({ langName, reviewedProgress })
  }
```

Reply:
It make sense!
Changes applied within [6c290e](https://github.com/badges/shields/pull/5472/commits/6c290e732a73ce2be8b163aca4a070f30e7dc741) commit.